File: CPP/Problems/remove-element.js
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function (nums, val) {
   
    let count = 0;
   
    for (let i = 0; i < nums.length; i++) {
        
        if (nums[i] !== val) {
            nums[count++] = nums[i];
        }
    }
    return count;
};
//Leetcode

File: CPP/arrays/RemoveElement.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        vector<int> :: iterator it;
        it = nums.begin();
        for(int i=0;i<nums.size();i++)
        {
            if(nums[i]==val)
            {
                nums.erase(it);
                it--;
                i--;
            }
            it++;
        }
        return nums.size();
    }
};